src/connectionController.ts

```diff
@@ -268,24 +269,50 @@ export default class ConnectionController {
     return connectionModel;
   };
 
-  public saveNewConnectionAndConnect = async (
-    connectionModel: ConnectionModelType,
-    connectionType: ConnectionTypes
-  ): Promise<ConnectionAttemptResult> => {
+  public getConnectionNameFromConnectionModel = (
+    connectionModel: ConnectionModelType
+  ): string => {
     const {
-      instanceId,
       sshTunnelOptions
     } = connectionModel.getAttributes({
       derived: true
     });
+
+    if (
+      connectionModel.sshTunnel &&
+      connectionModel.sshTunnel !== SSH_TUNNEL_TYPES.NONE &&
+      sshTunnelOptions.host &&
+      sshTunnelOptions.port
+    ) {
+      return `SSH tunnel to ${connectionModel.hosts.map(
+        ({ host, port }) => `${host}:${port}`
+      ).join(',')}`;
+    }
+
+    if (connectionModel.isSrvRecord) {
+      return connectionModel.hostname;
+    }
+
+    if (connectionModel.hosts && connectionModel.hosts.length > 0) {
+      return connectionModel.hosts.map(
+        ({ host, port }) => `${host}:${port}`
+      ).join(',');
+    }
+
+    return connectionModel.hostname;
```

Comment:
how is `hostname` used in VSCODE? don't we always have `hosts`?

Reply:
I think it's there to make sure we work nicely with `connection-model`.  Yes `hosts` should always be there and this line should never really be reached.

Reply:
Ok cool!